core/src/main/scala/tamer/Registry.scala

```diff
@@ -15,40 +14,44 @@ trait Registry {
 }
 
 object Registry {
-  final object Fake extends Registry {
-    override def getOrRegisterId(subject: String, schema: ParsedSchema): Task[Int] = UIO(-1)
-    override def verifySchema(id: Int, schema: ParsedSchema): Task[Unit]           = UIO.unit
+  final object FakeRegistry extends Registry {
+    override def getOrRegisterId(subject: String, schema: ParsedSchema): Task[Int] = ZIO.succeed(-1)
+    override def verifySchema(id: Int, schema: ParsedSchema): Task[Unit]           = ZIO.unit
   }
 
-  def live(url: String, size: Int = 1000, configuration: Map[String, Any] = Map.empty): Layer[TamerError, Has[Registry]] = ZLayer.fromEffect {
-    Task(new CachedSchemaRegistryClient(url, size, configuration.asJava)).mapError(TamerError("Cannot construct registry client", _)).map { client =>
-      new Registry {
-        private[this] final val logTask: Task[LogWriter[Task]] = log4sFromName.provide("tamer.registry")
-        private[this] final def getId(subject: String, schema: ParsedSchema, log: LogWriter[Task]): Task[Int] =
-          Task(client.getId(subject, schema)).tap(id => log.debug(s"retrieved existing writer schema id: $id"))
-        private[this] final def register(subject: String, schema: ParsedSchema, log: LogWriter[Task]): Task[Int] =
-          Task(client.register(subject, schema)).tap(id => log.info(s"registered with id $id new subject $subject writer schema $schema"))
-        private[this] final def get(id: Int, log: LogWriter[Task]): Task[ParsedSchema] =
-          Task(client.getSchemaById(id)).tap(_ => log.debug(s"retrieved writer schema id: $id"))
-        @nowarn private[this] final def verify(schema: ParsedSchema, writerSchema: ParsedSchema): Task[Unit] =
-          Task(schema.isBackwardCompatible(writerSchema).asScala)
-            .filterOrElse(_.isEmpty) { errors =>
-              ZIO.fail(TamerError(s"backwards incompatible schema: ${errors.mkString(", ")}"))
-            }
-            .unit
-
-        override final def getOrRegisterId(subject: String, schema: ParsedSchema): Task[Int] = for {
-          log <- logTask
-          id  <- getId(subject, schema, log) <> register(subject, schema, log)
-        } yield id
-        override final def verifySchema(id: Int, schema: ParsedSchema): Task[Unit] = for {
-          log          <- logTask
-          writerSchema <- get(id, log)
-          _            <- verify(schema, writerSchema)
-        } yield ()
-      }
-    }
+  final case class LiveRegistry(private val client: SchemaRegistryClient) extends Registry {
+    private[this] final val logTask: Task[LogWriter[Task]] = log4sFromName.provideEnvironment(ZEnvironment("tamer.LiveRegistry"))
+    private[this] final def getId(subject: String, schema: ParsedSchema, log: LogWriter[Task]): Task[Int] =
+      ZIO.attemptBlocking(client.getId(subject, schema)).tap(id => log.debug(s"retrieved existing writer schema id: $id"))
```

Comment:
`attemptBlocking` isn't due to migration requirements, is it? Is this an attempt to address the race condition issues we were facing with some registries?

Reply:
Apparently ZIO2 is capable of shifting automatically to the blocking pool when blocking IO is involved but I like being a bit more explicit. And, yes, this is a case where we need the blocking pool... see [here](https://github.com/confluentinc/schema-registry/blob/30d6677134fd88dece28cfafed5421ff5c7fc277/client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/RestService.java#L284-L343)